neurom/core/_soma.py

```diff
@@ -43,24 +42,38 @@ class Soma:
     and provides iterator access to them.
     """
 
-    def __init__(self, points):
-        """Initialize a Soma object."""
-        self._points = points
+    def __init__(self, morphio_soma):
+        """Constructor.
+
+        Args:
+            morphio_soma (morphio.Soma): instance of soma of MorphIO class
+        """
+        self._morphio_soma = morphio_soma
+        # this radius is used only for `volume` method, please avoid using it for anything else.
         self.radius = 0
 
     @property
     def center(self):
         """Obtain the center from the first stored point."""
-        return self._points[0][COLS.XYZ]
+        return self._morphio_soma.points[0]
```

Comment:
does this make any sense? Isn't the center the center of mass of all the points?

Reply:
ah, somasimplecontour does the center of mass, so this soma has  single point, or something like that.